### DISCUSSION TITLE
ESP8266 - update with minimal.bin.gz fails - not enough memory

### DISCUSSION THREAD

**Discussion by Trawacho** [arendst/Tasmota#19957] [Q&A]
Hi,
I´m not able to update to minimal.bin.gz version. 
Error:  "not enough space" or in german: "Ungenügend Speicherplatz".
As I understand it, the minimal version is also too big.

Information:
ESP Chip ID | 13668482 (ESP8266EX) 
Flash Chip Id | 0x187020 (DOUT) 
Flash Size | 16384 KB
Program Flash Size | 1024 KB
Program Size | 745 KB
Free Program Space | 256 KB
Free Memory | 18.0 KB

Is there a way to get a smaller minimal version, or any other idea?

**Comment by barbudor** [comment]
You seem to have a big tasmota variant. Self biult ?

There is no smaller version than Tasmota minimal 
So you may want to try that https://github.com/Jason2866/ESP8266MinimalHTTPUpdater

And if your Tamsota is a self built, I strongly suggest to shrink it down before upgrading again
